hdt-lib/src/hdt/BasicHDT.cpp

```diff
@@ -835,7 +838,7 @@ void BasicHDT::saveToHDT(std::ostream & output, ProgressListener *listener)
 
 void BasicHDT::loadOrCreateIndex(ProgressListener *listener) {
 
-	string indexname = this->fileName + ".index";
+	string indexname = this->fileName + ".index.v" + HDT_VERSION + INDEX_VERSION;
```

Comment:
If this concatenation occurs frequently, might be good to have a function / macro for it.